pkg/asset/installconfig/powervs/session.go

```diff
@@ -19,105 +24,208 @@ var (
 )
 
 // Session is an object representing a session for the IBM Power VS API.
+// A bluemix session object may be a better fit here
 type Session struct {
 	Session *ibmpisession.IBMPISession
 }
 
-// GetSession returns an IBM Cloud session by using credentials found in default locations in order:
-// env IBMID & env IBMID_PASSWORD,
-// ~/.bluemix/config.json ? (see TODO below)
-// and, if no creds are found, asks for them
-/* @TODO: if you do an `ibmcloud login` (or in my case ibmcloud login --sso), you get
-//  a very nice creds file at ~/.bluemix/config.json, with an IAMToken. There's no username,
-//  though (just the account's owner id, but that's not the same). It may be necessary
-//  to use the IAMToken vs the password env var mentioned here:
-//  https://github.com/IBM-Cloud/power-go-client#ibm-cloud-sdk-for-power-cloud
-//  Yes, I think we'll need to use the IAMToken. There's a two-factor auth built into the ibmcloud login,
-//  so the password alone isn't enough. The IAMToken is generated as a result. So either:
-     1) require the user has done this already and pull from the file
-     2) ask the user to paste in their IAMToken.
-     3) let the password env var be the IAMToken? (Going with this atm since it's how I started)
-     4) put it into Platform {userid: , iamtoken: , ...}
-*/
+// PISessionVars is an object that holds the variables required to create an ibmpisession object
+type PISessionVars struct {
+	ID     string `json:"id,omitempty"`
+	APIKey string `json:"apikey,omitempty"`
+	Region string `json:"region,omitempty"`
+	Zone   string `json:"zone,omitempty"`
+}
+
+// GetSession returns an ibmpisession object
 func GetSession() (*Session, error) {
 	s, err := getPISession()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to load credentials")
 	}
-
 	return &Session{Session: s}, nil
 }
 
-/*
-//  https://github.com/IBM-Cloud/power-go-client/blob/master/ibmpisession/ibmpowersession.go
-*/
+var (
+	defaultAuthFilePath    = filepath.Join(os.Getenv("HOME"), ".powervs", "config.json")
+	defaultBluemixFilePath = filepath.Join(os.Getenv("HOME"), ".bluemix", "config.json")
+)
+
 func getPISession() (*ibmpisession.IBMPISession, error) {
 
-	var (
-		id, iamtoken, apikey, region, zone string
-	)
+	var err error
+	var pisv PISessionVars
+
+	// First grab variables from the installer written authFilePath
+	logrus.Debug("Gathering variables from AuthFile")
+	err = getPISessionVarsFromAuthFile(&pisv)
+	if err != nil {
+		return nil, err
+	}
+
+	// Second grab variables from files
+
+	// .bluemix/config.json doesn't seem to hold anything useful but other files might
```

Comment:
let's remove this comment just b/c it's a bit too wordy.
but on the topic of the ~/.bluemix/config.json, it'll only contain something if a user intentionally created it and gave it that name (`ibmcloud iam api-key-create test --file ~/.bluemix/apikey_test.json --output json`).

Reply:
what do you think about just removing any checking for this file? it's not documented anywhere and i don't even remember where I found it when this project started. imo it's just a complication, and we can document that for IPI, users have to use the env vars.